rust/main/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -80,6 +91,39 @@ impl SealevelProvider {
         Ok(sender)
     }
 
+    fn recipient(hash: &H512, txn: &UiTransaction) -> ChainResult<H256> {
+        let message = Self::parsed_message(txn)?;
+
+        let programs = message
+            .instructions
+            .iter()
+            .filter_map(|ii| {
+                if let UiInstruction::Parsed(iii) = ii {
+                    Some(iii)
+                } else {
+                    None
+                }
+            })
+            .map(|ii| match ii {
+                UiParsedInstruction::Parsed(iii) => &iii.program_id,
+                UiParsedInstruction::PartiallyDecoded(iii) => &iii.program_id,
+            })
+            .filter(|program_id| !NATIVE_PROGRAMS.contains(*program_id))
+            .collect::<Vec<&String>>();
+
+        if programs.len() > 1 {
+            Err(HyperlaneSealevelError::TooManyNonNativePrograms(*hash))?;
```

Comment:
should we err here? I think it's decently likely that there could be multiple - I think it'd make sense to just grab the first one instead of erroring?

Reply:
We have considered it during adding Cosmos and settled on erroring. Getting an error here should give us a better idea how we should handle it.